src/webparts/sendToLaserficheLoginComponent/components/SendToLaserficheLoginComponent.tsx

```diff
@@ -11,641 +9,306 @@ import {
-        filelink = `https://app.${this.state.region}/laserfiche/DocView.aspx?db=${repoId}&docid=${Entryid6}`;
-
-        document.getElementById('it').innerHTML = 'Document uploaded';
-        document.getElementById('imgid').style.display = 'none';
-        document.getElementById('divid').style.display = 'block';
-        document.getElementById('divid1').onclick = this.Dc;
-        document.getElementById('divid13').style.display = 'block';
-        document.getElementById('divid13').onclick = this.viewfile;
-        document.getElementById('divid14').onclick = this.Redirect;
-        if (Action === ActionTypes.COPY) {
-          window.localStorage.removeItem(TempStorageKeys.LContType);
-        } else if (Action === ActionTypes.MOVE_AND_DELETE) {
-          this.DeleteFile(pageOrigin, Fileurl, Filenamewithext);
-        } else if (Action === ActionTypes.REPLACE) {
-          this.deletefileandreplace(
-            pageOrigin,
-            Fileurl,
-            Filenamewithoutext,
-            Filenamewithext,
-            filelink,
-            contextPageAbsoluteUrl
-          );
-        }
-      } catch (error) {
-        if (error.operations.setFields.exceptions[0].statusCode === 409) {
-          const entryidConflict4 = error.operations.entryCreate.entryId;
-          filelink = `https://app.${this.state.region}/laserfiche/DocView.aspx?db=${repoId}&docid=${entryidConflict4}`;
-
-          document.getElementById('it').innerHTML =
-            'Document uploaded to repository, updating metadata failed due to constraint mismatch<br/> <p style="color:red">The Laserfiche template and fields were not applied to this document</p>';
-          document.getElementById('imgid').style.display = 'none';
-          document.getElementById('divid').style.display = 'block';
-          document.getElementById('divid1').onclick = this.Dc;
-          document.getElementById('divid13').style.display = 'block';
-          document.getElementById('divid13').onclick = this.viewfile;
-          document.getElementById('divid14').onclick = this.Redirect;
-          window.localStorage.removeItem(TempStorageKeys.LContType);
-        } else {
-          window.alert(`Error uploding file: ${JSON.stringify(error)}`);
-          window.localStorage.removeItem(TempStorageKeys.LContType);
-          dialog.close();
-        }
+      if (spFileMetadata.action === ActionTypes.COPY) {
+        window.localStorage.removeItem('spdocdata');
+      } else if (spFileMetadata.action === ActionTypes.MOVE_AND_DELETE) {
+        DeleteFile(
+          spFileMetadata.pageOrigin,
+          spFileMetadata.fileUrl,
+          spFileMetadata.fileName
+        );
+      } else if (spFileMetadata.action === ActionTypes.REPLACE) {
+        deletefileandreplace(
+          spFileMetadata.pageOrigin,
+          spFileMetadata.fileUrl,
+          filenameWithoutExt,
+          spFileMetadata.fileName,
+          fileLink,
+          spFileMetadata.contextPageAbsoluteUrl
```

Comment:
should await deleteFileAndReplace and deleteFile.
if there is an action, does the API calls happen while the dialog is shown? does the dialog's inner text change depends on the API call's result?

Reply:
They were actually using alert for messages regarding the copy/delete.. This can be simplified after I refactor this code. So right now the user will be notified but it should be standardized